lib/std/crypto/aes.zig

```diff
@@ -9,9 +9,9 @@ const testing = std.testing;
 const builtin = std.builtin;
 
 pub const aes = comptime {
-    const has_aesni = builtin.Target.x86.featureSetHas(builtin.cpu.features, .aes);
-    const has_avx = builtin.Target.x86.featureSetHas(builtin.cpu.features, .avx);
-    const implementation = if (has_aesni and has_avx) @import("aes/aesni.zig") else @import("aes/soft.zig");
+    const has_aesni = comptime builtin.Target.x86.featureSetHas(builtin.cpu.features, .aes);
```

Comment:
Its already comptime, isn't it?

Reply:
I thought so as well, but CI disagrees :)

I guess `builtin.cpu.features` is comptime, but not `featureSetHas()`. Existing code also explicit mentions `comptime` (ex: `comptime builtin.arch.isMIPS()` in `os/linux.zig`).

Reply:
Damn. CI is still not happy.

In spite of the explicit `comptime`, Zig still tries to compile `aes/aesni.zig` on a `aarch64-linux-musl` target. Which of course fails since this is `x86_64` code.

Any idea why, anyone?

Reply:
[raw log](https://dev.azure.com/ziglang/0184afc2-b8d3-409b-8e4f-4032aaca4c96/_apis/build/builds/9233/logs/32) (in the end)
>2020-09-23T15:43:13.0134472Z <inline asm>:1:3: error: invalid instruction
2020-09-23T15:43:13.0136113Z          vaesenc q1, q0, q0
2020-09-23T15:43:13.0137371Z          ^
2020-09-23T15:43:13.0138076Z LLVM ERROR: Error parsing inline asm

Reply:
@jedisct1 I dont understand what you are trying to do with `comptime { return struct {} }` let it be simply a struct

Reply:
How does this even compile, i think return outside of a function should be a compile error

Reply:
@jedisct1 This is because you are not looking at the current Target, but just a target in general. you probably want to use `std.Target.current.cpu.arch == .x86_64`

Reply:
The implementation depends on the platform.

The issue here is that I expected

```text
if (comptime (has_aesni and has_avx)) @import("aes/aesni.zig") else @import("aes/soft.zig");
```

to not even try to compile `aes/aesni.zig` on a non-x86_64 platform. But it apparently does.

Reply:
Try this:
```zig
pub const aes = struct {
    const has_aesni = comptime builtin.Target.x86.featureSetHas(builtin.cpu.features, .aes);
    const has_avx = comptime builtin.Target.x86.featureSetHas(builtin.cpu.features, .avx);
    const impl = @import(if (has_aesni and has_avx) "aes/aesni.zig" else "aes/soft.zig");

    pub const Block = impl.Block;
    pub const AESEncryptCtx = impl.AESEncryptCtx;
    pub const AESDecryptCtx = impl.AESDecryptCtx;
    pub const AES128 = impl.AES128;
    pub const AES256 = impl.AES256;
};
```

Reply:
@FireFox317 Do you mean in addition to `builtin.cpu.features`?

What has to be tested here is a set of CPU features, not just the architecture.

Reply:
I should just first do a general check comptime check for the x86_64 arch and then in the if statement do the more generic feature set options.

Reply:
Success 👍 

Thanks a ton for your help, everyone! ❤️